paasta_tools/marathon_serviceinit.py

```diff
@@ -103,13 +103,29 @@ def status_desired_state(service, instance, client, job_config):
     return "State:      %s - Desired state: %s" % (status, desired_state)
 
 
+def get_app_queue_status(app_id, client):
+    app_queue = client.list_queue()
+    for app_queue_item in app_queue:
+        if app_queue_item.app.id == app_id:
+            return (app_queue_item.delay.overdue, app_queue_item.delay.timeLeftSeconds)
+
+    return (None, None)
+
+
 def status_marathon_job(service, instance, app_id, normal_instance_count, client):
     name = PaastaColors.cyan(compose_job_id(service, instance))
     if marathon_tools.is_app_id_running(app_id, client):
         app = client.get_app(app_id)
         running_instances = app.tasks_running
+        is_overdue, backoff_seconds = get_app_queue_status(app_id, client)
+
         if len(app.deployments) == 0:
```

Comment:
This now is too early, right? The order here will mater, this probably needs to go further down in the if clause